// @aws-sdk/middleware-endpoint-discovery/dist/es/getEndpointDiscoveryPlugin.js
import { __assign } from "tslib";
import { endpointDiscoveryMiddleware } from "./endpointDiscoveryMiddleware";
export var endpointDiscoveryMiddlewareOptions = {
    name: "endpointDiscoveryMiddleware",
    step: "build",
    tags: ["ENDPOINT_DISCOVERY"],
    override: true,
};
export var getEndpointDiscoveryPlugin = function (pluginConfig, middlewareConfig) { return ({
    applyToStack: function (commandStack) {
        commandStack.add(endpointDiscoveryMiddleware(pluginConfig, middlewareConfig), endpointDiscoveryMiddlewareOptions);
    },
}); };
export var getEndpointDiscoveryRequiredPlugin = function (pluginConfig, middlewareConfig) { return ({
    applyToStack: function (commandStack) {
        commandStack.add(endpointDiscoveryMiddleware(pluginConfig, __assign(__assign({}, middlewareConfig), { isDiscoveredEndpointRequired: true })), endpointDiscoveryMiddlewareOptions);
    },
}); };
export var getEndpointDiscoveryOptionalPlugin = function (pluginConfig, middlewareConfig) { return ({
    applyToStack: function (commandStack) {
        commandStack.add(endpointDiscoveryMiddleware(pluginConfig, __assign(__assign({}, middlewareConfig), { isDiscoveredEndpointRequired: false })), endpointDiscoveryMiddlewareOptions);
    },
}); };
