// @aws-sdk/client-dynamodb/dist/es/models/models_0.js
import { __assign, __read } from "tslib";
export var ArchivalSummary;
(function (ArchivalSummary) {
    /**
     * @internal
     */
    ArchivalSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ArchivalSummary || (ArchivalSummary = {}));
export var AttributeDefinition;
(function (AttributeDefinition) {
    /**
     * @internal
     */
    AttributeDefinition.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AttributeDefinition || (AttributeDefinition = {}));
export var AutoScalingTargetTrackingScalingPolicyConfigurationDescription;
(function (AutoScalingTargetTrackingScalingPolicyConfigurationDescription) {
    /**
     * @internal
     */
    AutoScalingTargetTrackingScalingPolicyConfigurationDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoScalingTargetTrackingScalingPolicyConfigurationDescription || (AutoScalingTargetTrackingScalingPolicyConfigurationDescription = {}));
export var AutoScalingPolicyDescription;
(function (AutoScalingPolicyDescription) {
    /**
     * @internal
     */
    AutoScalingPolicyDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoScalingPolicyDescription || (AutoScalingPolicyDescription = {}));
export var AutoScalingTargetTrackingScalingPolicyConfigurationUpdate;
(function (AutoScalingTargetTrackingScalingPolicyConfigurationUpdate) {
    /**
     * @internal
     */
    AutoScalingTargetTrackingScalingPolicyConfigurationUpdate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoScalingTargetTrackingScalingPolicyConfigurationUpdate || (AutoScalingTargetTrackingScalingPolicyConfigurationUpdate = {}));
export var AutoScalingPolicyUpdate;
(function (AutoScalingPolicyUpdate) {
    /**
     * @internal
     */
    AutoScalingPolicyUpdate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoScalingPolicyUpdate || (AutoScalingPolicyUpdate = {}));
export var AutoScalingSettingsDescription;
(function (AutoScalingSettingsDescription) {
    /**
     * @internal
     */
    AutoScalingSettingsDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoScalingSettingsDescription || (AutoScalingSettingsDescription = {}));
export var AutoScalingSettingsUpdate;
(function (AutoScalingSettingsUpdate) {
    /**
     * @internal
     */
    AutoScalingSettingsUpdate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(AutoScalingSettingsUpdate || (AutoScalingSettingsUpdate = {}));
export var BackupType;
(function (BackupType) {
    BackupType["AWS_BACKUP"] = "AWS_BACKUP";
    BackupType["SYSTEM"] = "SYSTEM";
    BackupType["USER"] = "USER";
})(BackupType || (BackupType = {}));
export var BackupDetails;
(function (BackupDetails) {
    /**
     * @internal
     */
    BackupDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BackupDetails || (BackupDetails = {}));
export var KeySchemaElement;
(function (KeySchemaElement) {
    /**
     * @internal
     */
    KeySchemaElement.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(KeySchemaElement || (KeySchemaElement = {}));
export var ProvisionedThroughput;
(function (ProvisionedThroughput) {
    /**
     * @internal
     */
    ProvisionedThroughput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ProvisionedThroughput || (ProvisionedThroughput = {}));
export var SourceTableDetails;
(function (SourceTableDetails) {
    /**
     * @internal
     */
    SourceTableDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SourceTableDetails || (SourceTableDetails = {}));
export var Projection;
(function (Projection) {
    /**
     * @internal
     */
    Projection.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Projection || (Projection = {}));
export var GlobalSecondaryIndexInfo;
(function (GlobalSecondaryIndexInfo) {
    /**
     * @internal
     */
    GlobalSecondaryIndexInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalSecondaryIndexInfo || (GlobalSecondaryIndexInfo = {}));
export var LocalSecondaryIndexInfo;
(function (LocalSecondaryIndexInfo) {
    /**
     * @internal
     */
    LocalSecondaryIndexInfo.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LocalSecondaryIndexInfo || (LocalSecondaryIndexInfo = {}));
export var SSEDescription;
(function (SSEDescription) {
    /**
     * @internal
     */
    SSEDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SSEDescription || (SSEDescription = {}));
export var StreamSpecification;
(function (StreamSpecification) {
    /**
     * @internal
     */
    StreamSpecification.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(StreamSpecification || (StreamSpecification = {}));
export var TimeToLiveDescription;
(function (TimeToLiveDescription) {
    /**
     * @internal
     */
    TimeToLiveDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TimeToLiveDescription || (TimeToLiveDescription = {}));
export var SourceTableFeatureDetails;
(function (SourceTableFeatureDetails) {
    /**
     * @internal
     */
    SourceTableFeatureDetails.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SourceTableFeatureDetails || (SourceTableFeatureDetails = {}));
export var BackupDescription;
(function (BackupDescription) {
    /**
     * @internal
     */
    BackupDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BackupDescription || (BackupDescription = {}));
export var BackupInUseException;
(function (BackupInUseException) {
    /**
     * @internal
     */
    BackupInUseException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BackupInUseException || (BackupInUseException = {}));
export var BackupNotFoundException;
(function (BackupNotFoundException) {
    /**
     * @internal
     */
    BackupNotFoundException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BackupNotFoundException || (BackupNotFoundException = {}));
export var BackupSummary;
(function (BackupSummary) {
    /**
     * @internal
     */
    BackupSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BackupSummary || (BackupSummary = {}));
export var BackupTypeFilter;
(function (BackupTypeFilter) {
    BackupTypeFilter["ALL"] = "ALL";
    BackupTypeFilter["AWS_BACKUP"] = "AWS_BACKUP";
    BackupTypeFilter["SYSTEM"] = "SYSTEM";
    BackupTypeFilter["USER"] = "USER";
})(BackupTypeFilter || (BackupTypeFilter = {}));
export var BatchStatementErrorCodeEnum;
(function (BatchStatementErrorCodeEnum) {
    BatchStatementErrorCodeEnum["AccessDenied"] = "AccessDenied";
    BatchStatementErrorCodeEnum["ConditionalCheckFailed"] = "ConditionalCheckFailed";
    BatchStatementErrorCodeEnum["DuplicateItem"] = "DuplicateItem";
    BatchStatementErrorCodeEnum["InternalServerError"] = "InternalServerError";
    BatchStatementErrorCodeEnum["ItemCollectionSizeLimitExceeded"] = "ItemCollectionSizeLimitExceeded";
    BatchStatementErrorCodeEnum["ProvisionedThroughputExceeded"] = "ProvisionedThroughputExceeded";
    BatchStatementErrorCodeEnum["RequestLimitExceeded"] = "RequestLimitExceeded";
    BatchStatementErrorCodeEnum["ResourceNotFound"] = "ResourceNotFound";
    BatchStatementErrorCodeEnum["ThrottlingError"] = "ThrottlingError";
    BatchStatementErrorCodeEnum["TransactionConflict"] = "TransactionConflict";
    BatchStatementErrorCodeEnum["ValidationError"] = "ValidationError";
})(BatchStatementErrorCodeEnum || (BatchStatementErrorCodeEnum = {}));
export var BatchStatementError;
(function (BatchStatementError) {
    /**
     * @internal
     */
    BatchStatementError.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BatchStatementError || (BatchStatementError = {}));
export var InternalServerError;
(function (InternalServerError) {
    /**
     * @internal
     */
    InternalServerError.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InternalServerError || (InternalServerError = {}));
export var RequestLimitExceeded;
(function (RequestLimitExceeded) {
    /**
     * @internal
     */
    RequestLimitExceeded.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RequestLimitExceeded || (RequestLimitExceeded = {}));
export var Capacity;
(function (Capacity) {
    /**
     * @internal
     */
    Capacity.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Capacity || (Capacity = {}));
export var ConsumedCapacity;
(function (ConsumedCapacity) {
    /**
     * @internal
     */
    ConsumedCapacity.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConsumedCapacity || (ConsumedCapacity = {}));
export var InvalidEndpointException;
(function (InvalidEndpointException) {
    /**
     * @internal
     */
    InvalidEndpointException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidEndpointException || (InvalidEndpointException = {}));
export var ProvisionedThroughputExceededException;
(function (ProvisionedThroughputExceededException) {
    /**
     * @internal
     */
    ProvisionedThroughputExceededException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ProvisionedThroughputExceededException || (ProvisionedThroughputExceededException = {}));
export var ResourceNotFoundException;
(function (ResourceNotFoundException) {
    /**
     * @internal
     */
    ResourceNotFoundException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResourceNotFoundException || (ResourceNotFoundException = {}));
export var ItemCollectionSizeLimitExceededException;
(function (ItemCollectionSizeLimitExceededException) {
    /**
     * @internal
     */
    ItemCollectionSizeLimitExceededException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ItemCollectionSizeLimitExceededException || (ItemCollectionSizeLimitExceededException = {}));
export var BillingModeSummary;
(function (BillingModeSummary) {
    /**
     * @internal
     */
    BillingModeSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(BillingModeSummary || (BillingModeSummary = {}));
export var ConditionalCheckFailedException;
(function (ConditionalCheckFailedException) {
    /**
     * @internal
     */
    ConditionalCheckFailedException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ConditionalCheckFailedException || (ConditionalCheckFailedException = {}));
export var PointInTimeRecoveryDescription;
(function (PointInTimeRecoveryDescription) {
    /**
     * @internal
     */
    PointInTimeRecoveryDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PointInTimeRecoveryDescription || (PointInTimeRecoveryDescription = {}));
export var ContinuousBackupsDescription;
(function (ContinuousBackupsDescription) {
    /**
     * @internal
     */
    ContinuousBackupsDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ContinuousBackupsDescription || (ContinuousBackupsDescription = {}));
export var ContinuousBackupsUnavailableException;
(function (ContinuousBackupsUnavailableException) {
    /**
     * @internal
     */
    ContinuousBackupsUnavailableException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ContinuousBackupsUnavailableException || (ContinuousBackupsUnavailableException = {}));
export var ContributorInsightsSummary;
(function (ContributorInsightsSummary) {
    /**
     * @internal
     */
    ContributorInsightsSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ContributorInsightsSummary || (ContributorInsightsSummary = {}));
export var CreateBackupInput;
(function (CreateBackupInput) {
    /**
     * @internal
     */
    CreateBackupInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateBackupInput || (CreateBackupInput = {}));
export var CreateBackupOutput;
(function (CreateBackupOutput) {
    /**
     * @internal
     */
    CreateBackupOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateBackupOutput || (CreateBackupOutput = {}));
export var LimitExceededException;
(function (LimitExceededException) {
    /**
     * @internal
     */
    LimitExceededException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LimitExceededException || (LimitExceededException = {}));
export var TableInUseException;
(function (TableInUseException) {
    /**
     * @internal
     */
    TableInUseException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TableInUseException || (TableInUseException = {}));
export var TableNotFoundException;
(function (TableNotFoundException) {
    /**
     * @internal
     */
    TableNotFoundException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TableNotFoundException || (TableNotFoundException = {}));
export var CreateGlobalSecondaryIndexAction;
(function (CreateGlobalSecondaryIndexAction) {
    /**
     * @internal
     */
    CreateGlobalSecondaryIndexAction.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGlobalSecondaryIndexAction || (CreateGlobalSecondaryIndexAction = {}));
export var Replica;
(function (Replica) {
    /**
     * @internal
     */
    Replica.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Replica || (Replica = {}));
export var CreateGlobalTableInput;
(function (CreateGlobalTableInput) {
    /**
     * @internal
     */
    CreateGlobalTableInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGlobalTableInput || (CreateGlobalTableInput = {}));
export var ProvisionedThroughputOverride;
(function (ProvisionedThroughputOverride) {
    /**
     * @internal
     */
    ProvisionedThroughputOverride.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ProvisionedThroughputOverride || (ProvisionedThroughputOverride = {}));
export var ReplicaGlobalSecondaryIndexDescription;
(function (ReplicaGlobalSecondaryIndexDescription) {
    /**
     * @internal
     */
    ReplicaGlobalSecondaryIndexDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicaGlobalSecondaryIndexDescription || (ReplicaGlobalSecondaryIndexDescription = {}));
export var ReplicaDescription;
(function (ReplicaDescription) {
    /**
     * @internal
     */
    ReplicaDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicaDescription || (ReplicaDescription = {}));
export var GlobalTableDescription;
(function (GlobalTableDescription) {
    /**
     * @internal
     */
    GlobalTableDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalTableDescription || (GlobalTableDescription = {}));
export var CreateGlobalTableOutput;
(function (CreateGlobalTableOutput) {
    /**
     * @internal
     */
    CreateGlobalTableOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateGlobalTableOutput || (CreateGlobalTableOutput = {}));
export var GlobalTableAlreadyExistsException;
(function (GlobalTableAlreadyExistsException) {
    /**
     * @internal
     */
    GlobalTableAlreadyExistsException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalTableAlreadyExistsException || (GlobalTableAlreadyExistsException = {}));
export var CreateReplicaAction;
(function (CreateReplicaAction) {
    /**
     * @internal
     */
    CreateReplicaAction.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateReplicaAction || (CreateReplicaAction = {}));
export var ReplicaGlobalSecondaryIndex;
(function (ReplicaGlobalSecondaryIndex) {
    /**
     * @internal
     */
    ReplicaGlobalSecondaryIndex.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicaGlobalSecondaryIndex || (ReplicaGlobalSecondaryIndex = {}));
export var CreateReplicationGroupMemberAction;
(function (CreateReplicationGroupMemberAction) {
    /**
     * @internal
     */
    CreateReplicationGroupMemberAction.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateReplicationGroupMemberAction || (CreateReplicationGroupMemberAction = {}));
export var GlobalSecondaryIndex;
(function (GlobalSecondaryIndex) {
    /**
     * @internal
     */
    GlobalSecondaryIndex.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalSecondaryIndex || (GlobalSecondaryIndex = {}));
export var LocalSecondaryIndex;
(function (LocalSecondaryIndex) {
    /**
     * @internal
     */
    LocalSecondaryIndex.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LocalSecondaryIndex || (LocalSecondaryIndex = {}));
export var SSESpecification;
(function (SSESpecification) {
    /**
     * @internal
     */
    SSESpecification.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(SSESpecification || (SSESpecification = {}));
export var Tag;
(function (Tag) {
    /**
     * @internal
     */
    Tag.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Tag || (Tag = {}));
export var CreateTableInput;
(function (CreateTableInput) {
    /**
     * @internal
     */
    CreateTableInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateTableInput || (CreateTableInput = {}));
export var ProvisionedThroughputDescription;
(function (ProvisionedThroughputDescription) {
    /**
     * @internal
     */
    ProvisionedThroughputDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ProvisionedThroughputDescription || (ProvisionedThroughputDescription = {}));
export var GlobalSecondaryIndexDescription;
(function (GlobalSecondaryIndexDescription) {
    /**
     * @internal
     */
    GlobalSecondaryIndexDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalSecondaryIndexDescription || (GlobalSecondaryIndexDescription = {}));
export var LocalSecondaryIndexDescription;
(function (LocalSecondaryIndexDescription) {
    /**
     * @internal
     */
    LocalSecondaryIndexDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(LocalSecondaryIndexDescription || (LocalSecondaryIndexDescription = {}));
export var RestoreSummary;
(function (RestoreSummary) {
    /**
     * @internal
     */
    RestoreSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RestoreSummary || (RestoreSummary = {}));
export var TableDescription;
(function (TableDescription) {
    /**
     * @internal
     */
    TableDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TableDescription || (TableDescription = {}));
export var CreateTableOutput;
(function (CreateTableOutput) {
    /**
     * @internal
     */
    CreateTableOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(CreateTableOutput || (CreateTableOutput = {}));
export var ResourceInUseException;
(function (ResourceInUseException) {
    /**
     * @internal
     */
    ResourceInUseException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ResourceInUseException || (ResourceInUseException = {}));
export var DeleteBackupInput;
(function (DeleteBackupInput) {
    /**
     * @internal
     */
    DeleteBackupInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteBackupInput || (DeleteBackupInput = {}));
export var DeleteBackupOutput;
(function (DeleteBackupOutput) {
    /**
     * @internal
     */
    DeleteBackupOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteBackupOutput || (DeleteBackupOutput = {}));
export var DeleteGlobalSecondaryIndexAction;
(function (DeleteGlobalSecondaryIndexAction) {
    /**
     * @internal
     */
    DeleteGlobalSecondaryIndexAction.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteGlobalSecondaryIndexAction || (DeleteGlobalSecondaryIndexAction = {}));
export var TransactionConflictException;
(function (TransactionConflictException) {
    /**
     * @internal
     */
    TransactionConflictException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TransactionConflictException || (TransactionConflictException = {}));
export var DeleteReplicaAction;
(function (DeleteReplicaAction) {
    /**
     * @internal
     */
    DeleteReplicaAction.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteReplicaAction || (DeleteReplicaAction = {}));
export var DeleteReplicationGroupMemberAction;
(function (DeleteReplicationGroupMemberAction) {
    /**
     * @internal
     */
    DeleteReplicationGroupMemberAction.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteReplicationGroupMemberAction || (DeleteReplicationGroupMemberAction = {}));
export var DeleteTableInput;
(function (DeleteTableInput) {
    /**
     * @internal
     */
    DeleteTableInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteTableInput || (DeleteTableInput = {}));
export var DeleteTableOutput;
(function (DeleteTableOutput) {
    /**
     * @internal
     */
    DeleteTableOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DeleteTableOutput || (DeleteTableOutput = {}));
export var DescribeBackupInput;
(function (DescribeBackupInput) {
    /**
     * @internal
     */
    DescribeBackupInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeBackupInput || (DescribeBackupInput = {}));
export var DescribeBackupOutput;
(function (DescribeBackupOutput) {
    /**
     * @internal
     */
    DescribeBackupOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeBackupOutput || (DescribeBackupOutput = {}));
export var DescribeContinuousBackupsInput;
(function (DescribeContinuousBackupsInput) {
    /**
     * @internal
     */
    DescribeContinuousBackupsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeContinuousBackupsInput || (DescribeContinuousBackupsInput = {}));
export var DescribeContinuousBackupsOutput;
(function (DescribeContinuousBackupsOutput) {
    /**
     * @internal
     */
    DescribeContinuousBackupsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeContinuousBackupsOutput || (DescribeContinuousBackupsOutput = {}));
export var DescribeContributorInsightsInput;
(function (DescribeContributorInsightsInput) {
    /**
     * @internal
     */
    DescribeContributorInsightsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeContributorInsightsInput || (DescribeContributorInsightsInput = {}));
export var FailureException;
(function (FailureException) {
    /**
     * @internal
     */
    FailureException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(FailureException || (FailureException = {}));
export var DescribeContributorInsightsOutput;
(function (DescribeContributorInsightsOutput) {
    /**
     * @internal
     */
    DescribeContributorInsightsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeContributorInsightsOutput || (DescribeContributorInsightsOutput = {}));
export var DescribeEndpointsRequest;
(function (DescribeEndpointsRequest) {
    /**
     * @internal
     */
    DescribeEndpointsRequest.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEndpointsRequest || (DescribeEndpointsRequest = {}));
export var Endpoint;
(function (Endpoint) {
    /**
     * @internal
     */
    Endpoint.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(Endpoint || (Endpoint = {}));
export var DescribeEndpointsResponse;
(function (DescribeEndpointsResponse) {
    /**
     * @internal
     */
    DescribeEndpointsResponse.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeEndpointsResponse || (DescribeEndpointsResponse = {}));
export var DescribeExportInput;
(function (DescribeExportInput) {
    /**
     * @internal
     */
    DescribeExportInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeExportInput || (DescribeExportInput = {}));
export var ExportFormat;
(function (ExportFormat) {
    ExportFormat["DYNAMODB_JSON"] = "DYNAMODB_JSON";
    ExportFormat["ION"] = "ION";
})(ExportFormat || (ExportFormat = {}));
export var ExportStatus;
(function (ExportStatus) {
    ExportStatus["COMPLETED"] = "COMPLETED";
    ExportStatus["FAILED"] = "FAILED";
    ExportStatus["IN_PROGRESS"] = "IN_PROGRESS";
})(ExportStatus || (ExportStatus = {}));
export var ExportDescription;
(function (ExportDescription) {
    /**
     * @internal
     */
    ExportDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ExportDescription || (ExportDescription = {}));
export var DescribeExportOutput;
(function (DescribeExportOutput) {
    /**
     * @internal
     */
    DescribeExportOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeExportOutput || (DescribeExportOutput = {}));
export var ExportNotFoundException;
(function (ExportNotFoundException) {
    /**
     * @internal
     */
    ExportNotFoundException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ExportNotFoundException || (ExportNotFoundException = {}));
export var DescribeGlobalTableInput;
(function (DescribeGlobalTableInput) {
    /**
     * @internal
     */
    DescribeGlobalTableInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeGlobalTableInput || (DescribeGlobalTableInput = {}));
export var DescribeGlobalTableOutput;
(function (DescribeGlobalTableOutput) {
    /**
     * @internal
     */
    DescribeGlobalTableOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeGlobalTableOutput || (DescribeGlobalTableOutput = {}));
export var GlobalTableNotFoundException;
(function (GlobalTableNotFoundException) {
    /**
     * @internal
     */
    GlobalTableNotFoundException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalTableNotFoundException || (GlobalTableNotFoundException = {}));
export var DescribeGlobalTableSettingsInput;
(function (DescribeGlobalTableSettingsInput) {
    /**
     * @internal
     */
    DescribeGlobalTableSettingsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeGlobalTableSettingsInput || (DescribeGlobalTableSettingsInput = {}));
export var ReplicaGlobalSecondaryIndexSettingsDescription;
(function (ReplicaGlobalSecondaryIndexSettingsDescription) {
    /**
     * @internal
     */
    ReplicaGlobalSecondaryIndexSettingsDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicaGlobalSecondaryIndexSettingsDescription || (ReplicaGlobalSecondaryIndexSettingsDescription = {}));
export var ReplicaSettingsDescription;
(function (ReplicaSettingsDescription) {
    /**
     * @internal
     */
    ReplicaSettingsDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicaSettingsDescription || (ReplicaSettingsDescription = {}));
export var DescribeGlobalTableSettingsOutput;
(function (DescribeGlobalTableSettingsOutput) {
    /**
     * @internal
     */
    DescribeGlobalTableSettingsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeGlobalTableSettingsOutput || (DescribeGlobalTableSettingsOutput = {}));
export var DescribeKinesisStreamingDestinationInput;
(function (DescribeKinesisStreamingDestinationInput) {
    /**
     * @internal
     */
    DescribeKinesisStreamingDestinationInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeKinesisStreamingDestinationInput || (DescribeKinesisStreamingDestinationInput = {}));
export var KinesisDataStreamDestination;
(function (KinesisDataStreamDestination) {
    /**
     * @internal
     */
    KinesisDataStreamDestination.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(KinesisDataStreamDestination || (KinesisDataStreamDestination = {}));
export var DescribeKinesisStreamingDestinationOutput;
(function (DescribeKinesisStreamingDestinationOutput) {
    /**
     * @internal
     */
    DescribeKinesisStreamingDestinationOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeKinesisStreamingDestinationOutput || (DescribeKinesisStreamingDestinationOutput = {}));
export var DescribeLimitsInput;
(function (DescribeLimitsInput) {
    /**
     * @internal
     */
    DescribeLimitsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeLimitsInput || (DescribeLimitsInput = {}));
export var DescribeLimitsOutput;
(function (DescribeLimitsOutput) {
    /**
     * @internal
     */
    DescribeLimitsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeLimitsOutput || (DescribeLimitsOutput = {}));
export var DescribeTableInput;
(function (DescribeTableInput) {
    /**
     * @internal
     */
    DescribeTableInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeTableInput || (DescribeTableInput = {}));
export var DescribeTableOutput;
(function (DescribeTableOutput) {
    /**
     * @internal
     */
    DescribeTableOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeTableOutput || (DescribeTableOutput = {}));
export var DescribeTableReplicaAutoScalingInput;
(function (DescribeTableReplicaAutoScalingInput) {
    /**
     * @internal
     */
    DescribeTableReplicaAutoScalingInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeTableReplicaAutoScalingInput || (DescribeTableReplicaAutoScalingInput = {}));
export var ReplicaGlobalSecondaryIndexAutoScalingDescription;
(function (ReplicaGlobalSecondaryIndexAutoScalingDescription) {
    /**
     * @internal
     */
    ReplicaGlobalSecondaryIndexAutoScalingDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicaGlobalSecondaryIndexAutoScalingDescription || (ReplicaGlobalSecondaryIndexAutoScalingDescription = {}));
export var ReplicaAutoScalingDescription;
(function (ReplicaAutoScalingDescription) {
    /**
     * @internal
     */
    ReplicaAutoScalingDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicaAutoScalingDescription || (ReplicaAutoScalingDescription = {}));
export var TableAutoScalingDescription;
(function (TableAutoScalingDescription) {
    /**
     * @internal
     */
    TableAutoScalingDescription.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TableAutoScalingDescription || (TableAutoScalingDescription = {}));
export var DescribeTableReplicaAutoScalingOutput;
(function (DescribeTableReplicaAutoScalingOutput) {
    /**
     * @internal
     */
    DescribeTableReplicaAutoScalingOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeTableReplicaAutoScalingOutput || (DescribeTableReplicaAutoScalingOutput = {}));
export var DescribeTimeToLiveInput;
(function (DescribeTimeToLiveInput) {
    /**
     * @internal
     */
    DescribeTimeToLiveInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeTimeToLiveInput || (DescribeTimeToLiveInput = {}));
export var DescribeTimeToLiveOutput;
(function (DescribeTimeToLiveOutput) {
    /**
     * @internal
     */
    DescribeTimeToLiveOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DescribeTimeToLiveOutput || (DescribeTimeToLiveOutput = {}));
export var KinesisStreamingDestinationInput;
(function (KinesisStreamingDestinationInput) {
    /**
     * @internal
     */
    KinesisStreamingDestinationInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(KinesisStreamingDestinationInput || (KinesisStreamingDestinationInput = {}));
export var KinesisStreamingDestinationOutput;
(function (KinesisStreamingDestinationOutput) {
    /**
     * @internal
     */
    KinesisStreamingDestinationOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(KinesisStreamingDestinationOutput || (KinesisStreamingDestinationOutput = {}));
export var DuplicateItemException;
(function (DuplicateItemException) {
    /**
     * @internal
     */
    DuplicateItemException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(DuplicateItemException || (DuplicateItemException = {}));
export var IdempotentParameterMismatchException;
(function (IdempotentParameterMismatchException) {
    /**
     * @internal
     */
    IdempotentParameterMismatchException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(IdempotentParameterMismatchException || (IdempotentParameterMismatchException = {}));
export var TransactionInProgressException;
(function (TransactionInProgressException) {
    /**
     * @internal
     */
    TransactionInProgressException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TransactionInProgressException || (TransactionInProgressException = {}));
export var ExportConflictException;
(function (ExportConflictException) {
    /**
     * @internal
     */
    ExportConflictException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ExportConflictException || (ExportConflictException = {}));
export var ExportTableToPointInTimeInput;
(function (ExportTableToPointInTimeInput) {
    /**
     * @internal
     */
    ExportTableToPointInTimeInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ExportTableToPointInTimeInput || (ExportTableToPointInTimeInput = {}));
export var ExportTableToPointInTimeOutput;
(function (ExportTableToPointInTimeOutput) {
    /**
     * @internal
     */
    ExportTableToPointInTimeOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ExportTableToPointInTimeOutput || (ExportTableToPointInTimeOutput = {}));
export var InvalidExportTimeException;
(function (InvalidExportTimeException) {
    /**
     * @internal
     */
    InvalidExportTimeException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidExportTimeException || (InvalidExportTimeException = {}));
export var PointInTimeRecoveryUnavailableException;
(function (PointInTimeRecoveryUnavailableException) {
    /**
     * @internal
     */
    PointInTimeRecoveryUnavailableException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PointInTimeRecoveryUnavailableException || (PointInTimeRecoveryUnavailableException = {}));
export var ListBackupsInput;
(function (ListBackupsInput) {
    /**
     * @internal
     */
    ListBackupsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListBackupsInput || (ListBackupsInput = {}));
export var ListBackupsOutput;
(function (ListBackupsOutput) {
    /**
     * @internal
     */
    ListBackupsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListBackupsOutput || (ListBackupsOutput = {}));
export var ListContributorInsightsInput;
(function (ListContributorInsightsInput) {
    /**
     * @internal
     */
    ListContributorInsightsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListContributorInsightsInput || (ListContributorInsightsInput = {}));
export var ListContributorInsightsOutput;
(function (ListContributorInsightsOutput) {
    /**
     * @internal
     */
    ListContributorInsightsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListContributorInsightsOutput || (ListContributorInsightsOutput = {}));
export var ListExportsInput;
(function (ListExportsInput) {
    /**
     * @internal
     */
    ListExportsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListExportsInput || (ListExportsInput = {}));
export var ExportSummary;
(function (ExportSummary) {
    /**
     * @internal
     */
    ExportSummary.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ExportSummary || (ExportSummary = {}));
export var ListExportsOutput;
(function (ListExportsOutput) {
    /**
     * @internal
     */
    ListExportsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListExportsOutput || (ListExportsOutput = {}));
export var ListGlobalTablesInput;
(function (ListGlobalTablesInput) {
    /**
     * @internal
     */
    ListGlobalTablesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListGlobalTablesInput || (ListGlobalTablesInput = {}));
export var GlobalTable;
(function (GlobalTable) {
    /**
     * @internal
     */
    GlobalTable.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalTable || (GlobalTable = {}));
export var ListGlobalTablesOutput;
(function (ListGlobalTablesOutput) {
    /**
     * @internal
     */
    ListGlobalTablesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListGlobalTablesOutput || (ListGlobalTablesOutput = {}));
export var ListTablesInput;
(function (ListTablesInput) {
    /**
     * @internal
     */
    ListTablesInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTablesInput || (ListTablesInput = {}));
export var ListTablesOutput;
(function (ListTablesOutput) {
    /**
     * @internal
     */
    ListTablesOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTablesOutput || (ListTablesOutput = {}));
export var ListTagsOfResourceInput;
(function (ListTagsOfResourceInput) {
    /**
     * @internal
     */
    ListTagsOfResourceInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsOfResourceInput || (ListTagsOfResourceInput = {}));
export var ListTagsOfResourceOutput;
(function (ListTagsOfResourceOutput) {
    /**
     * @internal
     */
    ListTagsOfResourceOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ListTagsOfResourceOutput || (ListTagsOfResourceOutput = {}));
export var RestoreTableFromBackupInput;
(function (RestoreTableFromBackupInput) {
    /**
     * @internal
     */
    RestoreTableFromBackupInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RestoreTableFromBackupInput || (RestoreTableFromBackupInput = {}));
export var RestoreTableFromBackupOutput;
(function (RestoreTableFromBackupOutput) {
    /**
     * @internal
     */
    RestoreTableFromBackupOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RestoreTableFromBackupOutput || (RestoreTableFromBackupOutput = {}));
export var TableAlreadyExistsException;
(function (TableAlreadyExistsException) {
    /**
     * @internal
     */
    TableAlreadyExistsException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TableAlreadyExistsException || (TableAlreadyExistsException = {}));
export var InvalidRestoreTimeException;
(function (InvalidRestoreTimeException) {
    /**
     * @internal
     */
    InvalidRestoreTimeException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(InvalidRestoreTimeException || (InvalidRestoreTimeException = {}));
export var RestoreTableToPointInTimeInput;
(function (RestoreTableToPointInTimeInput) {
    /**
     * @internal
     */
    RestoreTableToPointInTimeInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RestoreTableToPointInTimeInput || (RestoreTableToPointInTimeInput = {}));
export var RestoreTableToPointInTimeOutput;
(function (RestoreTableToPointInTimeOutput) {
    /**
     * @internal
     */
    RestoreTableToPointInTimeOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(RestoreTableToPointInTimeOutput || (RestoreTableToPointInTimeOutput = {}));
export var TagResourceInput;
(function (TagResourceInput) {
    /**
     * @internal
     */
    TagResourceInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TagResourceInput || (TagResourceInput = {}));
export var UntagResourceInput;
(function (UntagResourceInput) {
    /**
     * @internal
     */
    UntagResourceInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UntagResourceInput || (UntagResourceInput = {}));
export var PointInTimeRecoverySpecification;
(function (PointInTimeRecoverySpecification) {
    /**
     * @internal
     */
    PointInTimeRecoverySpecification.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(PointInTimeRecoverySpecification || (PointInTimeRecoverySpecification = {}));
export var UpdateContinuousBackupsInput;
(function (UpdateContinuousBackupsInput) {
    /**
     * @internal
     */
    UpdateContinuousBackupsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateContinuousBackupsInput || (UpdateContinuousBackupsInput = {}));
export var UpdateContinuousBackupsOutput;
(function (UpdateContinuousBackupsOutput) {
    /**
     * @internal
     */
    UpdateContinuousBackupsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateContinuousBackupsOutput || (UpdateContinuousBackupsOutput = {}));
export var UpdateContributorInsightsInput;
(function (UpdateContributorInsightsInput) {
    /**
     * @internal
     */
    UpdateContributorInsightsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateContributorInsightsInput || (UpdateContributorInsightsInput = {}));
export var UpdateContributorInsightsOutput;
(function (UpdateContributorInsightsOutput) {
    /**
     * @internal
     */
    UpdateContributorInsightsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateContributorInsightsOutput || (UpdateContributorInsightsOutput = {}));
export var ReplicaAlreadyExistsException;
(function (ReplicaAlreadyExistsException) {
    /**
     * @internal
     */
    ReplicaAlreadyExistsException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicaAlreadyExistsException || (ReplicaAlreadyExistsException = {}));
export var ReplicaNotFoundException;
(function (ReplicaNotFoundException) {
    /**
     * @internal
     */
    ReplicaNotFoundException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicaNotFoundException || (ReplicaNotFoundException = {}));
export var ReplicaUpdate;
(function (ReplicaUpdate) {
    /**
     * @internal
     */
    ReplicaUpdate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicaUpdate || (ReplicaUpdate = {}));
export var UpdateGlobalTableInput;
(function (UpdateGlobalTableInput) {
    /**
     * @internal
     */
    UpdateGlobalTableInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateGlobalTableInput || (UpdateGlobalTableInput = {}));
export var UpdateGlobalTableOutput;
(function (UpdateGlobalTableOutput) {
    /**
     * @internal
     */
    UpdateGlobalTableOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateGlobalTableOutput || (UpdateGlobalTableOutput = {}));
export var IndexNotFoundException;
(function (IndexNotFoundException) {
    /**
     * @internal
     */
    IndexNotFoundException.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(IndexNotFoundException || (IndexNotFoundException = {}));
export var GlobalTableGlobalSecondaryIndexSettingsUpdate;
(function (GlobalTableGlobalSecondaryIndexSettingsUpdate) {
    /**
     * @internal
     */
    GlobalTableGlobalSecondaryIndexSettingsUpdate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalTableGlobalSecondaryIndexSettingsUpdate || (GlobalTableGlobalSecondaryIndexSettingsUpdate = {}));
export var ReplicaGlobalSecondaryIndexSettingsUpdate;
(function (ReplicaGlobalSecondaryIndexSettingsUpdate) {
    /**
     * @internal
     */
    ReplicaGlobalSecondaryIndexSettingsUpdate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicaGlobalSecondaryIndexSettingsUpdate || (ReplicaGlobalSecondaryIndexSettingsUpdate = {}));
export var ReplicaSettingsUpdate;
(function (ReplicaSettingsUpdate) {
    /**
     * @internal
     */
    ReplicaSettingsUpdate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicaSettingsUpdate || (ReplicaSettingsUpdate = {}));
export var UpdateGlobalTableSettingsInput;
(function (UpdateGlobalTableSettingsInput) {
    /**
     * @internal
     */
    UpdateGlobalTableSettingsInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateGlobalTableSettingsInput || (UpdateGlobalTableSettingsInput = {}));
export var UpdateGlobalTableSettingsOutput;
(function (UpdateGlobalTableSettingsOutput) {
    /**
     * @internal
     */
    UpdateGlobalTableSettingsOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateGlobalTableSettingsOutput || (UpdateGlobalTableSettingsOutput = {}));
export var UpdateGlobalSecondaryIndexAction;
(function (UpdateGlobalSecondaryIndexAction) {
    /**
     * @internal
     */
    UpdateGlobalSecondaryIndexAction.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateGlobalSecondaryIndexAction || (UpdateGlobalSecondaryIndexAction = {}));
export var GlobalSecondaryIndexUpdate;
(function (GlobalSecondaryIndexUpdate) {
    /**
     * @internal
     */
    GlobalSecondaryIndexUpdate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalSecondaryIndexUpdate || (GlobalSecondaryIndexUpdate = {}));
export var UpdateReplicationGroupMemberAction;
(function (UpdateReplicationGroupMemberAction) {
    /**
     * @internal
     */
    UpdateReplicationGroupMemberAction.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateReplicationGroupMemberAction || (UpdateReplicationGroupMemberAction = {}));
export var ReplicationGroupUpdate;
(function (ReplicationGroupUpdate) {
    /**
     * @internal
     */
    ReplicationGroupUpdate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicationGroupUpdate || (ReplicationGroupUpdate = {}));
export var UpdateTableInput;
(function (UpdateTableInput) {
    /**
     * @internal
     */
    UpdateTableInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateTableInput || (UpdateTableInput = {}));
export var UpdateTableOutput;
(function (UpdateTableOutput) {
    /**
     * @internal
     */
    UpdateTableOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateTableOutput || (UpdateTableOutput = {}));
export var GlobalSecondaryIndexAutoScalingUpdate;
(function (GlobalSecondaryIndexAutoScalingUpdate) {
    /**
     * @internal
     */
    GlobalSecondaryIndexAutoScalingUpdate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(GlobalSecondaryIndexAutoScalingUpdate || (GlobalSecondaryIndexAutoScalingUpdate = {}));
export var ReplicaGlobalSecondaryIndexAutoScalingUpdate;
(function (ReplicaGlobalSecondaryIndexAutoScalingUpdate) {
    /**
     * @internal
     */
    ReplicaGlobalSecondaryIndexAutoScalingUpdate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicaGlobalSecondaryIndexAutoScalingUpdate || (ReplicaGlobalSecondaryIndexAutoScalingUpdate = {}));
export var ReplicaAutoScalingUpdate;
(function (ReplicaAutoScalingUpdate) {
    /**
     * @internal
     */
    ReplicaAutoScalingUpdate.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(ReplicaAutoScalingUpdate || (ReplicaAutoScalingUpdate = {}));
export var UpdateTableReplicaAutoScalingInput;
(function (UpdateTableReplicaAutoScalingInput) {
    /**
     * @internal
     */
    UpdateTableReplicaAutoScalingInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateTableReplicaAutoScalingInput || (UpdateTableReplicaAutoScalingInput = {}));
export var UpdateTableReplicaAutoScalingOutput;
(function (UpdateTableReplicaAutoScalingOutput) {
    /**
     * @internal
     */
    UpdateTableReplicaAutoScalingOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateTableReplicaAutoScalingOutput || (UpdateTableReplicaAutoScalingOutput = {}));
export var TimeToLiveSpecification;
(function (TimeToLiveSpecification) {
    /**
     * @internal
     */
    TimeToLiveSpecification.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(TimeToLiveSpecification || (TimeToLiveSpecification = {}));
export var UpdateTimeToLiveInput;
(function (UpdateTimeToLiveInput) {
    /**
     * @internal
     */
    UpdateTimeToLiveInput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateTimeToLiveInput || (UpdateTimeToLiveInput = {}));
export var UpdateTimeToLiveOutput;
(function (UpdateTimeToLiveOutput) {
    /**
     * @internal
     */
    UpdateTimeToLiveOutput.filterSensitiveLog = function (obj) { return (__assign({}, obj)); };
})(UpdateTimeToLiveOutput || (UpdateTimeToLiveOutput = {}));
export var AttributeValue;
(function (AttributeValue) {
    AttributeValue.visit = function (value, visitor) {
        if (value.S !== undefined)
            return visitor.S(value.S);
        if (value.N !== undefined)
            return visitor.N(value.N);
        if (value.B !== undefined)
            return visitor.B(value.B);
        if (value.SS !== undefined)
            return visitor.SS(value.SS);
        if (value.NS !== undefined)
            return visitor.NS(value.NS);
        if (value.BS !== undefined)
            return visitor.BS(value.BS);
        if (value.M !== undefined)
            return visitor.M(value.M);
        if (value.L !== undefined)
            return visitor.L(value.L);
        if (value.NULL !== undefined)
            return visitor.NULL(value.NULL);
        if (value.BOOL !== undefined)
            return visitor.BOOL(value.BOOL);
        return visitor._(value.$unknown[0], value.$unknown[1]);
    };
    /**
     * @internal
     */
    AttributeValue.filterSensitiveLog = function (obj) {
        var _a;
        if (obj.S !== undefined)
            return { S: obj.S };
        if (obj.N !== undefined)
            return { N: obj.N };
        if (obj.B !== undefined)
            return { B: obj.B };
        if (obj.SS !== undefined)
            return { SS: obj.SS };
        if (obj.NS !== undefined)
            return { NS: obj.NS };
        if (obj.BS !== undefined)
            return { BS: obj.BS };
        if (obj.M !== undefined)
            return {
                M: Object.entries(obj.M).reduce(function (acc, _a) {
                    var _b;
                    var _c = __read(_a, 2), key = _c[0], value = _c[1];
                    return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
                }, {}),
            };
        if (obj.L !== undefined)
            return { L: obj.L.map(function (item) { return AttributeValue.filterSensitiveLog(item); }) };
        if (obj.NULL !== undefined)
            return { NULL: obj.NULL };
        if (obj.BOOL !== undefined)
            return { BOOL: obj.BOOL };
        if (obj.$unknown !== undefined)
            return _a = {}, _a[obj.$unknown[0]] = "UNKNOWN", _a;
    };
})(AttributeValue || (AttributeValue = {}));
export var AttributeValueUpdate;
(function (AttributeValueUpdate) {
    /**
     * @internal
     */
    AttributeValueUpdate.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Value && { Value: AttributeValue.filterSensitiveLog(obj.Value) }))); };
})(AttributeValueUpdate || (AttributeValueUpdate = {}));
export var BatchStatementRequest;
(function (BatchStatementRequest) {
    /**
     * @internal
     */
    BatchStatementRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Parameters && { Parameters: obj.Parameters.map(function (item) { return AttributeValue.filterSensitiveLog(item); }) }))); };
})(BatchStatementRequest || (BatchStatementRequest = {}));
export var BatchStatementResponse;
(function (BatchStatementResponse) {
    /**
     * @internal
     */
    BatchStatementResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Item && {
        Item: Object.entries(obj.Item).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(BatchStatementResponse || (BatchStatementResponse = {}));
export var CancellationReason;
(function (CancellationReason) {
    /**
     * @internal
     */
    CancellationReason.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Item && {
        Item: Object.entries(obj.Item).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(CancellationReason || (CancellationReason = {}));
export var Condition;
(function (Condition) {
    /**
     * @internal
     */
    Condition.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.AttributeValueList && {
        AttributeValueList: obj.AttributeValueList.map(function (item) { return AttributeValue.filterSensitiveLog(item); }),
    }))); };
})(Condition || (Condition = {}));
export var DeleteRequest;
(function (DeleteRequest) {
    /**
     * @internal
     */
    DeleteRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Key && {
        Key: Object.entries(obj.Key).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(DeleteRequest || (DeleteRequest = {}));
export var ExecuteStatementInput;
(function (ExecuteStatementInput) {
    /**
     * @internal
     */
    ExecuteStatementInput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Parameters && { Parameters: obj.Parameters.map(function (item) { return AttributeValue.filterSensitiveLog(item); }) }))); };
})(ExecuteStatementInput || (ExecuteStatementInput = {}));
export var Get;
(function (Get) {
    /**
     * @internal
     */
    Get.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Key && {
        Key: Object.entries(obj.Key).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(Get || (Get = {}));
export var GetItemInput;
(function (GetItemInput) {
    /**
     * @internal
     */
    GetItemInput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Key && {
        Key: Object.entries(obj.Key).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(GetItemInput || (GetItemInput = {}));
export var GetItemOutput;
(function (GetItemOutput) {
    /**
     * @internal
     */
    GetItemOutput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Item && {
        Item: Object.entries(obj.Item).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(GetItemOutput || (GetItemOutput = {}));
export var ItemCollectionMetrics;
(function (ItemCollectionMetrics) {
    /**
     * @internal
     */
    ItemCollectionMetrics.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.ItemCollectionKey && {
        ItemCollectionKey: Object.entries(obj.ItemCollectionKey).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(ItemCollectionMetrics || (ItemCollectionMetrics = {}));
export var ItemResponse;
(function (ItemResponse) {
    /**
     * @internal
     */
    ItemResponse.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Item && {
        Item: Object.entries(obj.Item).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(ItemResponse || (ItemResponse = {}));
export var ParameterizedStatement;
(function (ParameterizedStatement) {
    /**
     * @internal
     */
    ParameterizedStatement.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Parameters && { Parameters: obj.Parameters.map(function (item) { return AttributeValue.filterSensitiveLog(item); }) }))); };
})(ParameterizedStatement || (ParameterizedStatement = {}));
export var PutRequest;
(function (PutRequest) {
    /**
     * @internal
     */
    PutRequest.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Item && {
        Item: Object.entries(obj.Item).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(PutRequest || (PutRequest = {}));
export var ExecuteStatementOutput;
(function (ExecuteStatementOutput) {
    /**
     * @internal
     */
    ExecuteStatementOutput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Items && {
        Items: obj.Items.map(function (item) {
            return Object.entries(item).reduce(function (acc, _a) {
                var _b;
                var _c = __read(_a, 2), key = _c[0], value = _c[1];
                return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
            }, {});
        }),
    }))); };
})(ExecuteStatementOutput || (ExecuteStatementOutput = {}));
export var KeysAndAttributes;
(function (KeysAndAttributes) {
    /**
     * @internal
     */
    KeysAndAttributes.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Keys && {
        Keys: obj.Keys.map(function (item) {
            return Object.entries(item).reduce(function (acc, _a) {
                var _b;
                var _c = __read(_a, 2), key = _c[0], value = _c[1];
                return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
            }, {});
        }),
    }))); };
})(KeysAndAttributes || (KeysAndAttributes = {}));
export var TransactGetItem;
(function (TransactGetItem) {
    /**
     * @internal
     */
    TransactGetItem.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Get && { Get: Get.filterSensitiveLog(obj.Get) }))); };
})(TransactGetItem || (TransactGetItem = {}));
export var BatchExecuteStatementInput;
(function (BatchExecuteStatementInput) {
    /**
     * @internal
     */
    BatchExecuteStatementInput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Statements && { Statements: obj.Statements.map(function (item) { return BatchStatementRequest.filterSensitiveLog(item); }) }))); };
})(BatchExecuteStatementInput || (BatchExecuteStatementInput = {}));
export var BatchExecuteStatementOutput;
(function (BatchExecuteStatementOutput) {
    /**
     * @internal
     */
    BatchExecuteStatementOutput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Responses && { Responses: obj.Responses.map(function (item) { return BatchStatementResponse.filterSensitiveLog(item); }) }))); };
})(BatchExecuteStatementOutput || (BatchExecuteStatementOutput = {}));
export var ExecuteTransactionInput;
(function (ExecuteTransactionInput) {
    /**
     * @internal
     */
    ExecuteTransactionInput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.TransactStatements && {
        TransactStatements: obj.TransactStatements.map(function (item) { return ParameterizedStatement.filterSensitiveLog(item); }),
    }))); };
})(ExecuteTransactionInput || (ExecuteTransactionInput = {}));
export var ExecuteTransactionOutput;
(function (ExecuteTransactionOutput) {
    /**
     * @internal
     */
    ExecuteTransactionOutput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Responses && { Responses: obj.Responses.map(function (item) { return ItemResponse.filterSensitiveLog(item); }) }))); };
})(ExecuteTransactionOutput || (ExecuteTransactionOutput = {}));
export var TransactGetItemsOutput;
(function (TransactGetItemsOutput) {
    /**
     * @internal
     */
    TransactGetItemsOutput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.Responses && { Responses: obj.Responses.map(function (item) { return ItemResponse.filterSensitiveLog(item); }) }))); };
})(TransactGetItemsOutput || (TransactGetItemsOutput = {}));
export var TransactionCanceledException;
(function (TransactionCanceledException) {
    /**
     * @internal
     */
    TransactionCanceledException.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.CancellationReasons && {
        CancellationReasons: obj.CancellationReasons.map(function (item) { return CancellationReason.filterSensitiveLog(item); }),
    }))); };
})(TransactionCanceledException || (TransactionCanceledException = {}));
export var BatchGetItemInput;
(function (BatchGetItemInput) {
    /**
     * @internal
     */
    BatchGetItemInput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.RequestItems && {
        RequestItems: Object.entries(obj.RequestItems).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = KeysAndAttributes.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(BatchGetItemInput || (BatchGetItemInput = {}));
export var ExpectedAttributeValue;
(function (ExpectedAttributeValue) {
    /**
     * @internal
     */
    ExpectedAttributeValue.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.Value && { Value: AttributeValue.filterSensitiveLog(obj.Value) })), (obj.AttributeValueList && {
        AttributeValueList: obj.AttributeValueList.map(function (item) { return AttributeValue.filterSensitiveLog(item); }),
    }))); };
})(ExpectedAttributeValue || (ExpectedAttributeValue = {}));
export var TransactGetItemsInput;
(function (TransactGetItemsInput) {
    /**
     * @internal
     */
    TransactGetItemsInput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.TransactItems && {
        TransactItems: obj.TransactItems.map(function (item) { return TransactGetItem.filterSensitiveLog(item); }),
    }))); };
})(TransactGetItemsInput || (TransactGetItemsInput = {}));
export var TransactWriteItemsOutput;
(function (TransactWriteItemsOutput) {
    /**
     * @internal
     */
    TransactWriteItemsOutput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.ItemCollectionMetrics && {
        ItemCollectionMetrics: Object.entries(obj.ItemCollectionMetrics).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = value.map(function (item) { return ItemCollectionMetrics.filterSensitiveLog(item); }), _b)));
        }, {}),
    }))); };
})(TransactWriteItemsOutput || (TransactWriteItemsOutput = {}));
export var ConditionCheck;
(function (ConditionCheck) {
    /**
     * @internal
     */
    ConditionCheck.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.Key && {
        Key: Object.entries(obj.Key).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.ExpressionAttributeValues && {
        ExpressionAttributeValues: Object.entries(obj.ExpressionAttributeValues).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(ConditionCheck || (ConditionCheck = {}));
export var Delete;
(function (Delete) {
    /**
     * @internal
     */
    Delete.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.Key && {
        Key: Object.entries(obj.Key).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.ExpressionAttributeValues && {
        ExpressionAttributeValues: Object.entries(obj.ExpressionAttributeValues).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(Delete || (Delete = {}));
export var Put;
(function (Put) {
    /**
     * @internal
     */
    Put.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.Item && {
        Item: Object.entries(obj.Item).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.ExpressionAttributeValues && {
        ExpressionAttributeValues: Object.entries(obj.ExpressionAttributeValues).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(Put || (Put = {}));
export var Update;
(function (Update) {
    /**
     * @internal
     */
    Update.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.Key && {
        Key: Object.entries(obj.Key).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.ExpressionAttributeValues && {
        ExpressionAttributeValues: Object.entries(obj.ExpressionAttributeValues).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(Update || (Update = {}));
export var DeleteItemOutput;
(function (DeleteItemOutput) {
    /**
     * @internal
     */
    DeleteItemOutput.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.Attributes && {
        Attributes: Object.entries(obj.Attributes).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.ItemCollectionMetrics && {
        ItemCollectionMetrics: ItemCollectionMetrics.filterSensitiveLog(obj.ItemCollectionMetrics),
    }))); };
})(DeleteItemOutput || (DeleteItemOutput = {}));
export var PutItemOutput;
(function (PutItemOutput) {
    /**
     * @internal
     */
    PutItemOutput.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.Attributes && {
        Attributes: Object.entries(obj.Attributes).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.ItemCollectionMetrics && {
        ItemCollectionMetrics: ItemCollectionMetrics.filterSensitiveLog(obj.ItemCollectionMetrics),
    }))); };
})(PutItemOutput || (PutItemOutput = {}));
export var QueryOutput;
(function (QueryOutput) {
    /**
     * @internal
     */
    QueryOutput.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.Items && {
        Items: obj.Items.map(function (item) {
            return Object.entries(item).reduce(function (acc, _a) {
                var _b;
                var _c = __read(_a, 2), key = _c[0], value = _c[1];
                return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
            }, {});
        }),
    })), (obj.LastEvaluatedKey && {
        LastEvaluatedKey: Object.entries(obj.LastEvaluatedKey).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(QueryOutput || (QueryOutput = {}));
export var ScanOutput;
(function (ScanOutput) {
    /**
     * @internal
     */
    ScanOutput.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.Items && {
        Items: obj.Items.map(function (item) {
            return Object.entries(item).reduce(function (acc, _a) {
                var _b;
                var _c = __read(_a, 2), key = _c[0], value = _c[1];
                return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
            }, {});
        }),
    })), (obj.LastEvaluatedKey && {
        LastEvaluatedKey: Object.entries(obj.LastEvaluatedKey).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(ScanOutput || (ScanOutput = {}));
export var UpdateItemOutput;
(function (UpdateItemOutput) {
    /**
     * @internal
     */
    UpdateItemOutput.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.Attributes && {
        Attributes: Object.entries(obj.Attributes).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.ItemCollectionMetrics && {
        ItemCollectionMetrics: ItemCollectionMetrics.filterSensitiveLog(obj.ItemCollectionMetrics),
    }))); };
})(UpdateItemOutput || (UpdateItemOutput = {}));
export var WriteRequest;
(function (WriteRequest) {
    /**
     * @internal
     */
    WriteRequest.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.PutRequest && { PutRequest: PutRequest.filterSensitiveLog(obj.PutRequest) })), (obj.DeleteRequest && { DeleteRequest: DeleteRequest.filterSensitiveLog(obj.DeleteRequest) }))); };
})(WriteRequest || (WriteRequest = {}));
export var BatchGetItemOutput;
(function (BatchGetItemOutput) {
    /**
     * @internal
     */
    BatchGetItemOutput.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.Responses && {
        Responses: Object.entries(obj.Responses).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = value.map(function (item) {
                return Object.entries(item).reduce(function (acc, _a) {
                    var _b;
                    var _c = __read(_a, 2), key = _c[0], value = _c[1];
                    return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
                }, {});
            }), _b)));
        }, {}),
    })), (obj.UnprocessedKeys && {
        UnprocessedKeys: Object.entries(obj.UnprocessedKeys).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = KeysAndAttributes.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(BatchGetItemOutput || (BatchGetItemOutput = {}));
export var ScanInput;
(function (ScanInput) {
    /**
     * @internal
     */
    ScanInput.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign(__assign({}, obj), (obj.ScanFilter && {
        ScanFilter: Object.entries(obj.ScanFilter).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = Condition.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.ExclusiveStartKey && {
        ExclusiveStartKey: Object.entries(obj.ExclusiveStartKey).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.ExpressionAttributeValues && {
        ExpressionAttributeValues: Object.entries(obj.ExpressionAttributeValues).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(ScanInput || (ScanInput = {}));
export var BatchWriteItemInput;
(function (BatchWriteItemInput) {
    /**
     * @internal
     */
    BatchWriteItemInput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.RequestItems && {
        RequestItems: Object.entries(obj.RequestItems).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = value.map(function (item) { return WriteRequest.filterSensitiveLog(item); }), _b)));
        }, {}),
    }))); };
})(BatchWriteItemInput || (BatchWriteItemInput = {}));
export var DeleteItemInput;
(function (DeleteItemInput) {
    /**
     * @internal
     */
    DeleteItemInput.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign(__assign({}, obj), (obj.Key && {
        Key: Object.entries(obj.Key).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.Expected && {
        Expected: Object.entries(obj.Expected).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = ExpectedAttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.ExpressionAttributeValues && {
        ExpressionAttributeValues: Object.entries(obj.ExpressionAttributeValues).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(DeleteItemInput || (DeleteItemInput = {}));
export var PutItemInput;
(function (PutItemInput) {
    /**
     * @internal
     */
    PutItemInput.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign(__assign({}, obj), (obj.Item && {
        Item: Object.entries(obj.Item).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.Expected && {
        Expected: Object.entries(obj.Expected).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = ExpectedAttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.ExpressionAttributeValues && {
        ExpressionAttributeValues: Object.entries(obj.ExpressionAttributeValues).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(PutItemInput || (PutItemInput = {}));
export var QueryInput;
(function (QueryInput) {
    /**
     * @internal
     */
    QueryInput.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign(__assign(__assign({}, obj), (obj.KeyConditions && {
        KeyConditions: Object.entries(obj.KeyConditions).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = Condition.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.QueryFilter && {
        QueryFilter: Object.entries(obj.QueryFilter).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = Condition.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.ExclusiveStartKey && {
        ExclusiveStartKey: Object.entries(obj.ExclusiveStartKey).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.ExpressionAttributeValues && {
        ExpressionAttributeValues: Object.entries(obj.ExpressionAttributeValues).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(QueryInput || (QueryInput = {}));
export var BatchWriteItemOutput;
(function (BatchWriteItemOutput) {
    /**
     * @internal
     */
    BatchWriteItemOutput.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign({}, obj), (obj.UnprocessedItems && {
        UnprocessedItems: Object.entries(obj.UnprocessedItems).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = value.map(function (item) { return WriteRequest.filterSensitiveLog(item); }), _b)));
        }, {}),
    })), (obj.ItemCollectionMetrics && {
        ItemCollectionMetrics: Object.entries(obj.ItemCollectionMetrics).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = value.map(function (item) { return ItemCollectionMetrics.filterSensitiveLog(item); }), _b)));
        }, {}),
    }))); };
})(BatchWriteItemOutput || (BatchWriteItemOutput = {}));
export var UpdateItemInput;
(function (UpdateItemInput) {
    /**
     * @internal
     */
    UpdateItemInput.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign(__assign(__assign({}, obj), (obj.Key && {
        Key: Object.entries(obj.Key).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.AttributeUpdates && {
        AttributeUpdates: Object.entries(obj.AttributeUpdates).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValueUpdate.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.Expected && {
        Expected: Object.entries(obj.Expected).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = ExpectedAttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    })), (obj.ExpressionAttributeValues && {
        ExpressionAttributeValues: Object.entries(obj.ExpressionAttributeValues).reduce(function (acc, _a) {
            var _b;
            var _c = __read(_a, 2), key = _c[0], value = _c[1];
            return (__assign(__assign({}, acc), (_b = {}, _b[key] = AttributeValue.filterSensitiveLog(value), _b)));
        }, {}),
    }))); };
})(UpdateItemInput || (UpdateItemInput = {}));
export var TransactWriteItem;
(function (TransactWriteItem) {
    /**
     * @internal
     */
    TransactWriteItem.filterSensitiveLog = function (obj) { return (__assign(__assign(__assign(__assign(__assign({}, obj), (obj.ConditionCheck && { ConditionCheck: ConditionCheck.filterSensitiveLog(obj.ConditionCheck) })), (obj.Put && { Put: Put.filterSensitiveLog(obj.Put) })), (obj.Delete && { Delete: Delete.filterSensitiveLog(obj.Delete) })), (obj.Update && { Update: Update.filterSensitiveLog(obj.Update) }))); };
})(TransactWriteItem || (TransactWriteItem = {}));
export var TransactWriteItemsInput;
(function (TransactWriteItemsInput) {
    /**
     * @internal
     */
    TransactWriteItemsInput.filterSensitiveLog = function (obj) { return (__assign(__assign({}, obj), (obj.TransactItems && {
        TransactItems: obj.TransactItems.map(function (item) { return TransactWriteItem.filterSensitiveLog(item); }),
    }))); };
})(TransactWriteItemsInput || (TransactWriteItemsInput = {}));
